addons/pos_pinelabs/static/src/app/utils/payment_pinelabs.js

```diff
@@ -0,0 +1,212 @@
+import { _t } from "@web/core/l10n/translation";
+import { PaymentInterface } from "@point_of_sale/app/payment/payment_interface";
+import { AlertDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { serializeDateTime } from "@web/core/l10n/dates";
+import { registry } from "@web/core/registry";
+
+const CANCEL_REQUEST_TIME_LIMIT = 125000;
+const { DateTime } = luxon;
+
+export class PaymentPinelabs extends PaymentInterface {
+    setup() {
+        super.setup(...arguments);
+        this.paymentInitialTime = null;
+        this.paymentLineResolvers = {};
+        this.offlineErrorHandler = registry.category("error_handlers").get("offlineErrorHandler");
+    }
+
+    send_payment_request(cid) {
+        super.send_payment_request(cid);
+        return this._processPinelabs();
+    }
+
+    pendingPinelabsPaymentLine() {
+        return this.pos.getPendingPaymentLine("pinelabs");
+    }
+
+    send_payment_cancel(order, cid) {
+        super.send_payment_cancel(order, cid);
+        return this._pinelabsCancel();
+    }
+
+    _callPinelabs(data, action) {
+        return this.pos.data
+            .silentCall("pos.payment.method", action, [[this.payment_method_id.id], data])
+            .catch((error) => {
+                const line = this.pendingPinelabsPaymentLine();
+                if (line) {
+                    line.set_payment_status("retry");
+                }
+                this.offlineErrorHandler(this.env, error, error);
+            });
```

Comment:
i know that in `pos_adyen` the same thing is done, but having `catch` after `silentCall` makes no sense
https://github.com/odoo/odoo/issues/new?permalink=https%3A%2F%2Fgithub.com%2Fodoo%2Fodoo%2Fblob%2Fcf1815ac93067b81be31b393d4930d6095275192%2Faddons%2Fpoint_of_sale%2Fstatic%2Fsrc%2Fapp%2Fservices%2Fdata_service.js%23L637-L645

Reply:
Can I use catch when calling a `call` method?

https://github.com/odoo/odoo/blob/cf1815ac93067b81be31b393d4930d6095275192/addons/point_of_sale/static/src/app/services/data_service.js#L633-L635

Reply:
yes, but keep in mind that not all errors will be allowed by `execute` to propagate
https://github.com/odoo/odoo/blob/cf1815ac93067b81be31b393d4930d6095275192/addons/point_of_sale/static/src/app/services/data_service.js#L462-L483